csrc/ops.h

```diff
@@ -183,6 +183,12 @@ void cutlass_moe_mm(
     torch::Tensor const& problem_sizes, torch::Tensor const& a_strides,
     torch::Tensor const& b_strides, torch::Tensor const& c_strides);
 
+void cutlass_moe_mm_fp16(
+    torch::Tensor& out_tensors, torch::Tensor const& a_tensors,
+    torch::Tensor const& b_tensors, torch::Tensor const& expert_offsets,
+    torch::Tensor const& problem_sizes, torch::Tensor const& a_strides,
+    torch::Tensor const& b_strides, torch::Tensor const& c_strides);
```

Comment:
I think the interface would be nicer if we just call it `cutlass_moe_mm`, and either do function overloading or make the `a_scales` and `b_scales` optional

Reply:
This would let us push the dtype dispatching into the C++ code, which would let us simplify the changes to `layer.py`

Reply:
I've merged them into one `cutlass_moe_mm` function.